cpp/src/Slice/Parser.cpp

```diff
@@ -522,47 +522,88 @@ Slice::DocComment::parseFrom(const ContainedPtr& p, DocLinkFormatter linkFormatt
+                    const string msg = "ignoring duplicate doc-comment tag: '" + paramTag + " " + name + "'";
+                    p->unit()->warning(p->file(), p->line(), InvalidComment, msg);
+                    currentSection = nullptr;
+                }
+                else
+                {
+                    comment._parameters[name] = {};
+                    currentSection = &(comment._parameters[name]);
+                }
             }
         }
-        else if (parseNamedCommentLine(l, exceptionTag, name, lineText))
+        else if (
+            parseNamedCommentLine(l, throwsTag, name, lineText) ||
+            parseNamedCommentLine(l, exceptionTag, name, lineText))
         {
-            if (!isOperation)
+            // '@throws' and '@exception' are equivalent. But we want to use the correct one in our warning messages.
+            const string actualTag = (l.find(throwsTag) == 0) ? throwsTag : exceptionTag;
+            if (!operationTarget)
             {
-                // If '@exception' was put on anything other than an operation, ignore it and issue a warning.
-                string msg = "the '" + exceptionTag + "' tag is only valid on operations";
+                // If '@throws'/'@exception' was put on anything other than an operation, ignore it and issue a warning.
+                const string msg = "the '" + actualTag + "' tag is only valid on operations";
                 p->unit()->warning(p->file(), p->line(), InvalidComment, msg);
                 currentSection = nullptr;
             }
             else
             {
-                comment._exceptions[name] = {};
-                currentSection = &(comment._exceptions[name]);
+                // Check if the exception exists...
+                const ExceptionPtr exceptionTarget = operationTarget->lookupException(name, false);
+                if (!exceptionTarget)
+                {
+                    const string msg = "'" + actualTag + " " + name +
+                                       "': no exception with this name could be found from the current scope";
+                    p->unit()->warning(p->file(), p->line(), InvalidComment, msg);
+                }
+                else
+                {
+                    // ... and matches one of the exceptions in the operation's specification.
+                    const ExceptionList exceptionSpec = operationTarget->throws();
+                    const auto exceptionCheck = [&exceptionTarget](const ExceptionPtr& ex)
+                    { return ex->isBaseOf(exceptionTarget) || ex->scoped() == exceptionTarget->scoped(); };
+                    if (std::find_if(exceptionSpec.begin(), exceptionSpec.end(), exceptionCheck) == exceptionSpec.end())
+                    {
+                        const string msg = "'" + actualTag + " " + name + "': this exception is not listed in " +
+                                           "(or a sub-exception of) this operation's exception specification";
```

Comment:
Maybe something like:

const string msg = "'" + actualTag + " " + name +
    "': this exception is neither listed in nor derived from any exception in the operation's exception specification";

Reply:
I have taken this.